### ISSUE TITLE
[FR] Make source control able to detect git bare repositories

### ISSUE THREAD

**Issue by sarpik** [microsoft/vscode#80946]
Hello,

TL;DR:

As far as I'm aware, `vscode` does not detect git **bare** repositories, and it'd be useful if it did.

---

I have a dotfiles setup as described in https://wiki.archlinux.org/index.php/Dotfiles#Tracking_dotfiles_directly_with_Git:

```sh
$ git init --bare ~/.dotfiles
$ alias config='/usr/bin/git --git-dir=$HOME/.dotfiles/ --work-tree=$HOME'
$ config config status.showUntrackedFiles no
```

Now, I often merge changes from upstream, and conflicts do come up.
I just like to handle merging with `vscode`, but so far I haven't been able to do this with my dotfile setup.

Is there something I'm missing, or is there a feature missing that would allow `vscode` to be aware of git bare repositories and / or git with `--git-dir` and `--work-dir` options?

Much thanks!

**Comment by joaomoreno** [comment]
> As far as I'm aware, vscode does not detect git bare repositories, and it'd be useful if it did.

Why would it be useful?

**Comment by sarpik** [comment]
@joaomoreno Hello, umm, I'm pretty sure I layed out my use case right below the issue's description. Expanding on it:

TL;DR:

What I want to do with `vscode` is committing changes & managing merge conflicts from upstream. That's it.

---

Currently I *can* go and manually open files with changes / merge conflicts.
But if `vscode` could detect a bare git repo, then I'd just point `vscode` to it and all the changes would be automatically available in `vscode`.

To clarify, I have a dotfiles setup in `$HOME` directory, tracked by `git`, without having the `.git` folder as `$HOME/.git`, but rather as a bare repository as, say, `$HOME/.dotfiles`, and using the `alias config='/usr/bin/git --git-dir=$HOME/.dotfiles/ --work-tree=$HOME'` to manage the dotfiles.

This provides a lot of advantages -- launching `git` will not work by default, only with the `config` alias, so you won't accidentally make a commit in the wrong repository if you have other repos in `$HOME/*`;
other `git` commands also won't do/affect anything by accident, since you're not using the `config` alias. If you are - you explicitly know that you're working with the dotfiles setup. More info available @ the [arch-wiki](https://wiki.archlinux.org/index.php/Dotfiles#Tracking_dotfiles_directly_with_Git)

> I have a dotfiles setup as described in [wiki.archlinux.org/index.php/Dotfiles#Tracking_dotfiles_directly_with_Git](https://wiki.archlinux.org/index.php/Dotfiles#Tracking_dotfiles_directly_with_Git):
> 
> ```shell
> $ git init --bare ~/.dotfiles
> $ alias config='/usr/bin/git --git-dir=$HOME/.dotfiles/ --work-tree=$HOME'
> $ config config status.showUntrackedFiles no
> ```
> 
> Now, I often merge changes from upstream, and conflicts do come up.
> I just like to handle merging with `vscode`, but so far I haven't been able to do this with my dotfile setup.
>

**Comment by joaomoreno** [comment]
Got it.

/duplicate https://github.com/microsoft/vscode/issues/77215

**Comment by sarpik** [comment]
Whoops, wasn't able to find the issue before. Any plans with it anytime soon?
